module-check/src/test/clojure/clojure/core/typed/test/testnew.clj

```diff
@@ -0,0 +1,55 @@
+(ns clojure.core.typed.test.testnew
+  (:require  [clojure.core.typed :as t] 
+             [clojure.test :refer :all]                
+             [clojure.core.typed.test.test-utils :refer :all]
+             [clojure.core.typed.test.destructure]))
+
+(deftest function?-test
+  (is-tc-e   (function? function?) Boolean
+             :requires [[clojure.test :refer [function?]]])
+  (is-tc-err   (function? function?) String
+             :requires [[clojure.test :refer [function?]]]))
+  
+(deftest assert-any-test
+  (is-tc-e   (assert-any "Hi" (= 4(+ 2 2) )) Any
+             :requires [[clojure.test :refer [assert-any]]]))
+           
+(deftest do-report-test
+  (is-tc-e  (do-report 1)  Any
+             :requires [[clojure.test :refer [do-report]]]))
+            
+(deftest run-tests-test
+  	(is-tc-e (defn a [] (run-tests)) (Map Any Any)
+             :requires [[clojure.test :refer [run-tests]]]))
+        
+(deftest run-all-tests-test
+  	(is-tc-e (defn a [] (run-all-tests)) (Map Any Any)
+             :requires [[clojure.test :refer [run-all-tests]]]))
+            
+(deftest successful?-test
+  (is-tc-e  (successful? 1) Boolean
+             :requires [[clojure.test :refer [successful?]]]))
+            
+(deftest compose-fixtures-test
+  	(is-tc-e (compose-fixtures (defn a [](+ 2 1)) (defn c [](+ 1 2) ) ) Any
+             :requires [[clojure.test :refer [compose-fixtures]]]))
```

Comment:
Ah. Actually this will work because `defn` returns a var, which is a function.

But I don't think we should use `defn` in nested contexts. Replace the `defn`s with `fn`s.